src/foam/nanos/notification/email/DAOResourceLoader.java

```diff
@@ -0,0 +1,42 @@
+package foam.nanos.notification.email;
+
+import com.google.common.base.Optional;
+import foam.dao.DAO;
+import org.jtwig.resource.loader.ResourceLoader;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public class DAOResourceLoader
+    implements ResourceLoader
+{
+  protected DAO dao_;
+
+  public DAOResourceLoader(DAO dao) {
+    this.dao_ = dao;
+  }
+
+  @Override
+  public Optional<Charset> getCharset(String s) {
+    return Optional.absent();
+  }
+
+  @Override
+  public InputStream load(String s) {
+    EmailTemplate template = (EmailTemplate) dao_.find(s);
+    return new ByteArrayInputStream(template.getBody().getBytes(StandardCharsets.UTF_8));
```

Comment:
Wouldn't it be more direct to just use StringBufferInputStream?

Reply:
StringBufferInputStream is deprecated and the Java docs say this in regards to it:
>This class does not properly convert characters into bytes.